### PR TITLE
Replace PushStage-based Log with GraphStage (part of #19834)

### PR DESCRIPTION
**Pull request by miciek**
Part of #19834.

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

File: akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala [hunk 1]
```diff
@@ -906,67 +906,86 @@ private[akka] final case class MapAsyncUnordered[In, Out](parallelism: Int, f: I
...
-  import Log._
-
-  private var logLevels: LogLevels = _
-  private var log: LoggingAdapter = _
+  override def toString = "Log"
 
   // TODO more optimisations can be done here - prepare logOnPush function etc
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) with OutHandler with InHandler {
+      override def toString = "LogLogic"
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala]: toString not needed for the logic AFAICS

Comment (miciek) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala]: Done.

**Comment by ktoso** [comment]
LGTM

**Comment by patriknw** [comment]
LGTM

**Pull request merged**